### ISSUE TITLE
evict leader timeout during rolling restart

### ISSUE THREAD

**Issue by youjiali1995** [tikv/tikv#10942]
### What version of TiKV are you using?

v4.0.11

### What operating system and CPU are you using?

doesn't matter.

### Steps to reproduce

Rolling restart the cluster.

### What did you expect?

Evict all leaders quickly.

### What did happened?

TiUP is waitting for transfering out leader until timeout.
![image](https://user-images.githubusercontent.com/14819777/133080149-c9b39b03-69c1-4c53-8318-43ddf70d658a.png)

From PD and TiKV's log, we found there is no transfer-leader operator since `2021/09/13 08:45:20.436` and the TiKV restarted at `2021/09/13 09:30:02.871`:
```
[2021/09/13 08:45:20.436 +00:00] [INFO] [operator_controller.go:537] ["operator finish"] [region-id=46312418] [takes=29.294468ms] [operator="\"evict-leader {transfer leader: store 1503111 to 349969} (kind:leader, region:46312418(37425,6195), createAt:2021-09-13 08:45:20.407355743 +0000 UTC m=+10109031.298382069, startAt:2021-09-13 08:45:20.407449644 +0000 UTC m=+10109031.298475965, currentStep:1, steps:[transfer leader from store 1503111 to store 349969]) finished\""] ["additional info"=]

[2021/09/13 08:45:20.419 +00:00] [INFO] [peer.rs:2109] ["transfer leader"] [peer="id: 46312421 store_id: 349969"] [peer_id=46312420] [region_id=46312418]
[2021/09/13 08:45:20.419 +00:00] [INFO] [raft.rs:1376] ["[term 9] starts to transfer leadership to 46312421"] [lead_transferee=46312421] [term=9] [raft_id=46312420] [region_id=46312418]

[2021/09/13 09:30:02.871 +00:00] [INFO] [lib.rs:92] ["Welcome to TiKV"]
```

TiUP gets the leader count of TiKV from metrics https://github.com/pingcap/tiup/pull/762 and the metric is calculated by `RegionCollector`: https://github.com/tikv/tikv/blob/b2117e26df80e1d4d85fcedbc7c6249ffda789c2/components/raftstore/src/coprocessor/region_info_accessor.rs#L489-L491

From PD's metrics, there is no leader on this store. I guess `RegionCollector` has bugs.
![image](https://user-images.githubusercontent.com/14819777/133081648-22427584-0591-44b8-8ec8-d77df9de1a5f.png)

**Comment by nolouch** [comment]
Why this issue is critical?

**Comment by youjiali1995** [comment]
> Why this issue is critical?
> 
> BTW, can I access the environment?

No, it's user's environment. I think the best way to find the root cause is reading the code.

**Comment by tonyxuqqi** [comment]
close it now unless reproed again.
